packages/models-library/src/models_library/service_settings_labels.py

```diff
@@ -0,0 +1,211 @@
+    __root__: List[SimcoreServiceSettingLabelEntry]
+
+    def __iter__(self):
+        return iter(self.__root__)
+
+    def __getitem__(self, item):
+        return self.__root__[item]
+
+
+class PathsMappingLabel(BaseModel):
+    inputs_path: Path = Field(
+        ..., description="path where the service expects all the inputs folder"
+    )
+    outputs_path: Path = Field(
+        ..., description="path where the service expects all the outputs folder"
+    )
+    state_paths: List[Path] = Field(
+        [],
+        description="optional list of path which contents need to be saved and restored",
+    )
+
+    @validator("state_paths", always=True)
+    @classmethod
+    def convert_none_to_empty_list(cls, v):
+        return [] if v is None else v
+
+    class Config(_BaseConfig):
+        schema_extra = {
+            "examples": {
+                "outputs_path": "/tmp/outputs",  # nosec
+                "inputs_path": "/tmp/inputs",  # nosec
+                "state_paths": ["/tmp/save_1", "/tmp_save_2"],  # nosec
+            }
+        }
+
+
+ComposeSpecLabel = Optional[Dict[str, Any]]
+
+
+class SimcoreServiceLabels(BaseModel):
+    """Validate all the simcores.services.* labels on a service"""
+
+    settings: Json[SimcoreServiceSettingsLabel] = Field(
+        ...,
+        alias="simcore.service.settings",
+        description=(
+            "Contains setting like environment variables and "
+            "resource constraints which are required by the service"
+        ),
+    )
+
+    paths_mapping: Json[Optional[PathsMappingLabel]] = Field(
+        None,
+        alias="simcore.service.paths-mapping",
+        description="json encoded, determines where the outputs and inputs directories are",
+    )
+
+    compose_spec: Json[ComposeSpecLabel] = Field(
+        None,
+        alias="simcore.service.compose-spec",
+        description="json encoded docker-compose spec",
+    )
+    container_http_entry: Optional[str] = Field(
+        None,
+        alias="simcore.service.container-http-entrypoint",
+        description=(
+            "When a compose spec is provided, a container where the proxy "
+            "needs to send http traffic must be specified"
+        ),
+    )
+
+    needs_dynamic_sidecar: Optional[bool] = Field(
+        None, description="true if dynamic sidecar is needed to start that service"
+    )
+
+    @validator("needs_dynamic_sidecar", pre=True, always=True)
+    @classmethod
+    def auto_fill_dynamic_sidecar(cls, v, values):
+        if not v:
+            return values.get("paths_mapping") is not None
+        return v
+
+    @validator("container_http_entry", always=True)
+    @classmethod
+    def compose_spec_requires_container_http_entry(cls, v, values):
+        if v is None and values.get("compose_spec") is not None:
+            raise ValueError(
+                "Field `container_http_entry` must be defined but is missing"
+            )
+        return v
+
```

Comment:
While trying to write a validator for this. I've realised this is already guaranteed. Will not add such a validator as it will make it more confusing. Instead I will add a comment about this behaviour.